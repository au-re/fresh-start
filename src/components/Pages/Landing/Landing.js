import React, { Component } from "react";
import styled from "styled-components";

import { fadeColor, fadeIn } from "../../../styles/animations";

const LandingTitle = styled.h1`
  color: white;
  align-self: center;
  text-transform: uppercase;
  opacity: 0;
  animation: ${fadeIn()} 0.5s ease-out forwards;
  animation-delay: 0.5s;
`;

const LoadingLanding = styled.div`
  background: salmon;
`;

/* the landing page */

class Landing extends Component {

  static Title = LandingTitle;
  static LoadingLanding = LoadingLanding;

  render() {
    const { className } = this.props;

    return (
      <div className={className}>
        <LandingTitle>I'm just a Boilerplate</LandingTitle>
      </div>);
  }
}

export default styled(Landing) `
  background: salmon;
  color: white;
  height: 100%;
  display: flex;
  justify-content: center;
  animation: ${fadeColor("salmon", "tomato")} 1s ease-out forwards;
`;
