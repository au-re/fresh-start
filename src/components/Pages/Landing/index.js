import React, { Component } from "react";

import Button from "../../Button";

class Landing extends Component {
  render() {
    const { className } = this.props;
    return (
      <div className={className}>
        landing page
        <Button>hello world</Button>
      </div>);
  }
}

export default Landing;
