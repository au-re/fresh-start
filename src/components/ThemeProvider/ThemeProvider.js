import PropTypes from "prop-types";
import React, { Component } from "react";
import { ThemeProvider as Provider } from "styled-components";

// TODO: move to component library

/* adds a function to set the theme on top of styled components themeprovider */

class ThemeProvider extends Component {

  state = {
    themes: this.props.themes,
    selectedTheme: this.props.themes[0],
  }

  static defaultProps = {
    themes: [{}],
  }

  static childContextTypes = {
    themes: PropTypes.array.isRequired,
    selectedTheme: PropTypes.object.isRequired,
    setTheme: PropTypes.func.isRequired,
  }

  getChildContext = () => {
    const { themes, selectedTheme } = this.state;
    return { themes, setTheme: this.setTheme, selectedTheme };
  }

  setTheme = (theme) => {
    this.setState({ selectedTheme: theme });
  }

  render() {
    return <Provider theme={this.state.selectedTheme}>
      {this.props.children}
    </Provider>;
  }
}

/* wrapper for any component that wishes to set the theme */

const withTheme = (ComponentToWrap) => class extends Component {

  static contextTypes = {
    themes: PropTypes.array.isRequired,
    setTheme: PropTypes.func.isRequired,
    selectedTheme: PropTypes.object.isRequired,
  }

  render() {
    const { themes, setTheme, selectedTheme } = this.context;
    return (<ComponentToWrap
      {...this.props}
      setTheme={setTheme}
      themes={themes}
      selectedTheme={selectedTheme} />);
  }
};

export default ThemeProvider;
export { withTheme };
