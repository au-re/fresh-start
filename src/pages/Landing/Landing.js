import get from "lodash.get";
import React, { Component } from "react";
import Helmet from "react-helmet";
import styled from "styled-components";

import GithubCorner from "../../components/GithubCorner/GithubCorner";
import ThemeMenu from "../../components/ThemeMenu/ThemeMenu";
import config from "../../config";
import { fadeBackgroundColor, fadeIn } from "../../styles";

const LandingTitle = styled.h1`
  color: ${({ theme }) => get(theme, "blends.landing.text")};
  font-weight: bold;
  font-size: 1.5rem;
  align-self: center;
  text-align: center;
  text-transform: uppercase;
  opacity: 0;
  animation: ${fadeIn()} 0.5s ease-out forwards;
  animation-delay: 0.5s;
`;

const LandingMenu = styled.div`
  align-self: center;
`;

const LandingFooter = styled.div`
  position: absolute;
  padding: 1rem;
  bottom: 0;
  left: 0;
`;

/* the landing page */

class Landing extends Component {

  render() {
    const { className } = this.props;

    return (
      <div className={className}>
        <Helmet>
          <title>Landing</title>
        </Helmet>
        <GithubCorner />
        <LandingTitle>I'm just a Boilerplate</LandingTitle>
        <LandingMenu>
          themeable
          <ThemeMenu />
        </LandingMenu>
        <LandingFooter>{config.copyrightNotice}<strong>{config.company}</strong></LandingFooter>
      </div>);
  }
}

export default styled(Landing)`
  background: ${({ theme }) => get(theme, "blends.landing.background")};
  color: ${({ theme }) => get(theme, "blends.landing.color")};
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  animation: ${({ theme }) => fadeBackgroundColor(
    get(theme, "blends.landing.backgroundIntro"),
    get(theme, "blends.landing.background"))} 1s ease-out forwards;
`;
