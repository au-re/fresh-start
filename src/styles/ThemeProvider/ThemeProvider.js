import PropTypes from "prop-types";
import React, { Component } from "react";
import { ThemeProvider as Provider } from "styled-components";

import pop from "../themes/pop";
import random from "../themes/random";

/* provides a function to set the theme */

class ThemeProvider extends Component {

  state = {
    themes: [pop, random],
    selected: pop,
  }

  static childContextTypes = {
    themes: PropTypes.object.isRequired,
    setTheme: PropTypes.func.isRequired,
  }

  getChildContext = () => {
    const { themes } = this.props;
    return { themes, setTheme: this.setTheme };
  }

  setTheme = (theme) => {
    this.setState({ selected: theme });
  }

  render() {
    return <Provider theme={this.state.selected}>
      {this.props.children}
    </Provider>;
  }
}

export default ThemeProvider;
