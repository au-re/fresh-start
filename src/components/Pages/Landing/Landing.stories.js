import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Landing from "./Landing";

storiesOf("Landing", module)

  .add("basic usage", withInfo(`A landing page`)(
    () => <Landing />));
