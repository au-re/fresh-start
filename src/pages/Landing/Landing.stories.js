import { storiesOf } from "@storybook/react";
import React from "react";
import { MemoryRouter as Router } from "react-router";

import ThemeProvider from "../../components/ThemeProvider/ThemeProvider";
import themes from "../../themes";
import Landing from "./Landing";

storiesOf("Landing Page", module)
  .add("basic usage", () =>
    <Router>
      <ThemeProvider themes={themes}>
        <Landing />
      </ThemeProvider>
    </Router>);
