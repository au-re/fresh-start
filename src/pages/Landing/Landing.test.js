import "jest-styled-components";

import React from "react";
import { MemoryRouter as Router } from "react-router";
import renderer from "react-test-renderer";

import ThemeProvider from "../../components/ThemeProvider/ThemeProvider";
import Landing from "./Landing";

describe("Landing", () => {

  it("renders without props", () => {
    const tree = renderer.create(
      <Router>
        <ThemeProvider>
          <Landing />
        </ThemeProvider>
      </Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
