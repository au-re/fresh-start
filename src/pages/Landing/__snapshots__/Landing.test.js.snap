// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders without props 1`] = `
.c1:hover .octo-arm {
  -webkit-animation: jJWalO 560ms ease-in-out;
  animation: jJWalO 560ms ease-in-out;
}

.c5 {
  min-width: 120px;
  font-size: 1rem;
  cursor: pointer;
}

.c4 {
  border-left: 1px solid;
  padding: 0 24px;
  margin: 0 24px;
  display: inline-block;
}

.c2 {
  font-weight: bold;
  font-size: 1.5rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  text-align: center;
  text-transform: uppercase;
  opacity: 0;
  -webkit-animation: bBzLxR 0.5s ease-out forwards;
  animation: bBzLxR 0.5s ease-out forwards;
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.c3 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 {
  position: absolute;
  padding: 1rem;
  bottom: 0;
  left: 0;
}

.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-animation: euXcxa 1s ease-out forwards;
  animation: euXcxa 1s ease-out forwards;
}

<div
  className="c0"
>
  <a
    className="c1"
    href="/"
  >
    <svg
      fill="#FFF"
      height={80}
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": "0",
          "zIndex": 10,
        }
      }
      viewBox="0 0 250 250"
      width={80}
    >
      <path
        d="M0 0l115 115h15l12 27 108 108V0z"
        fill="#151513"
      />
      <path
        className="octo-arm"
        d="M128 109c-15-9-9-19-9-19 3-7 2-11 2-11-1-7 3-2 3-2 4 5 2 11 2 11-3 10 5 15 9 16"
        style={
          Object {
            "WebkitTransformOrigin": "130px 106px",
            "transformOrigin": "130px 106px",
          }
        }
      />
      <path
        className="octo-body"
        d="M115 115s4 2 5 0l14-14c3-2 6-3 8-3-8-11-15-24 2-41 5-5 10-7 16-7 1-2 3-7 12-11 0 0 5 3 7 16 4 2 8 5 12 9s7 8 9 12c14 3 17 7 17 7-4 8-9 11-11 11 0 6-2 11-7 16-16 16-30 10-41 2 0 3-1 7-5 11l-12 11c-1 1 1 5 1 5z"
      />
    </svg>
  </a>
  <h1
    className="c2"
  >
    I'm just a Boilerplate
  </h1>
  <div
    className="c3"
  >
    themeable
    <div
      className="c4"
      style={undefined}
    >
      <div
        className="c5"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="c6"
  >
    Â© 2018 
    <strong>
      Acme Inc
    </strong>
  </div>
</div>
`;
